Nos/Controller/NoteEditorController.swift

```diff
@@ -0,0 +1,256 @@
+    private func insertMention(of author: Author, at range: NSRange) {
+        guard let textView, let url = author.uri else {
+            return
+        }
+        
+        let rangeIncludingAmpersand = NSRange(
+            location: max(range.location - 1, 0), 
+            length: min(range.length + 1, textView.attributedText.length)
+        )
+        insert(text: "@\(author.safeName)", link: url.absoluteString, at: rangeIncludingAmpersand)
+    }
+    
+    /// Handles the user pasting an npub at the given range.
+    private func insertMention(npub: String, range: NSRange) {
+        insert(text: npub.prefix(10).appending("..."), link: "nostr:\(npub)", at: range)
+    }
+    
+    /// Handles the user pasting an note link at the given range. 
+    private func insertMention(note: String, range: NSRange) {
+        insert(text: note.prefix(10).appending("..."), link: "nostr:\(note)", at: range)
+    }
+    
+    /// Inserts a string at the given range and adds the link attribute with the given `link`.
+    private func insert(text: String, link: String, at range: NSRange) {
+        guard let textView else {
+            return
+        }
+        
+        let linkAttributes = defaultNSAttributes.merging(
+            [NSAttributedString.Key.link: link], 
+            uniquingKeysWith: { _, key in key }
+        )
+        let link = NSAttributedString(
+            string: text,
+            attributes: linkAttributes
+        )
+        
+        let attributedString = textView.attributedText.mutableCopy() as! NSMutableAttributedString
+        attributedString.replaceCharacters(in: range, with: link)
+        textView.attributedText = attributedString
+        textView.selectedRange.location = range.location + link.length
+    }
+    
+    /// Takes the same arguments as `textView(_:shouldChangeTextIn:replacementText:)` and detects the case where the 
+    /// user is typing inside a link (like a mention of another author). If this is the case we remove the link 
+    /// attribute from the link and insert the given `newText` and return true. Otherwise this will return `false`.
+    private func removeLinkAttributesUnderCursor(
+        selectedRange: NSRange, 
+        in textView: UITextView, 
+        newText: String
+    ) -> Bool {
+        let attributedText = textView.attributedText!
+        if attributedText.length > 0, selectedRange.location < attributedText.length || newText.isEmpty {
+            var rangeOfLink = NSRange()
+            let location = newText.isEmpty ? max(selectedRange.location, 0) : selectedRange.location
+            let link = attributedText.attribute(
+                .link, 
+                at: location, 
+                longestEffectiveRange: &rangeOfLink, 
+                in: NSRange(location: 0, length: attributedText.length)
+            )
+            
+            if link != nil {
+                let newAttributesString = (attributedText.mutableCopy() as! NSMutableAttributedString)
```

Comment:
Same point as above. Does the [init](https://developer.apple.com/documentation/foundation/nsattributedstring/1415342-init) work for this?

```suggestion
                let newAttributesString = NSMutableAttributedString(attributedString: attributedText)
```

Reply:
Oh yeah, it does.